specifyweb/frontend/js_src/lib/components/DataModel/collection.ts

```diff
@@ -107,7 +107,7 @@ function mapValue(
     return value === true &&
       (tableName === 'Attachment' || typeof scopingField === 'object')
       ? 'true'
-      : undefined;
+      : (value as string).toString();
```

Comment:
hmm, this doesn't look right
the condition above `value === true && (tableName === 'Attachment' || typeof scopingField === 'object')` makes the return of this function true if `value` is true AND tableName or scopingField match some value

after your code change, if that condition doesn't match the function would still return 'true' even if the tableName or scopingField condition didn't match (because of `(value as string).toString()`), so you essentially made all of this redundant: `value === true && (tableName === 'Attachment' || typeof scopingField === 'object') ? 'true' :`
(which is incorrect, as that code as there for a reason)

in the other case, when `!value`, your code `(value as string).toString()` would return `'false'`, but that should be treated by the back-end same as committing domainFilter (the undefined value) - if back-end treats those a different than that might be a back-end bug

so that I can better understand the changes you are making, can you explain to me what is the bug that is causing https://github.com/specify/specify7/issues/4520 and how you are trying to solve it? or, if you didn't yet find the cause, what you tried/whether you need help